vspace/src/main/java/edu/asu/diging/vspace/web/exception/GlobalExceptionHandler.java

```
@@ -0,0 +1,79 @@
+        modelAndView.addObject("error", ex.getMessage());
+        logger.info("SlideNotFoundException Occured:: URL=" + request.getRequestURL());
+        modelAndView.addObject("url", request.getRequestURL());
+        modelAndView.setViewName("module");
+        return modelAndView;
+    }
+
+    @ExceptionHandler({ SequenceNotFoundException.class })
+    protected ModelAndView handleSequenceNotFoundException(HttpServletRequest request, SequenceNotFoundException ex) {
+        ModelAndView modelAndView = new ModelAndView();
+        modelAndView.addObject("error", ex.getMessage());
+        logger.info("SequenceNotFoundException Occured:: URL=" + request.getRequestURL());
+        modelAndView.addObject("url", request.getRequestURL());
+        modelAndView.setViewName("module");
+        return modelAndView;
+    }
+
+    @ExceptionHandler({ SlidesInSequenceNotFoundException.class })
+    protected ModelAndView handleSlidesInSequenceNotFoundException(HttpServletRequest request,
+            SlidesInSequenceNotFoundException ex) {
+        ModelAndView modelAndView = new ModelAndView();
+        modelAndView.addObject("error", ex.getMessage());
+        logger.info("SlidesInSequenceNotFoundException Occured:: URL=" + request.getRequestURL());
+        modelAndView.addObject("url", request.getRequestURL());
+        modelAndView.setViewName("module");
+        return modelAndView;
+    }
+
+    @ExceptionHandler({ SpaceNotFoundException.class })
+    protected ModelAndView handleSpaceNotFoundException(HttpServletRequest request,
+            SlidesInSequenceNotFoundException ex) {
+        ModelAndView modelAndView = new ModelAndView();
+        modelAndView.addObject("error", ex.getMessage());
+        logger.info("SpaceNotFoundException Occured:: URL=" + request.getRequestURL());
+        modelAndView.addObject("url", request.getRequestURL());
+        modelAndView.setViewName("module");
+        return modelAndView;
+    }
+
+}
```

you're swallowing the exceptions; they need to be printed in the logs.